wrapanapi/msazure.py

```diff
@@ -299,13 +299,34 @@ def get_vm_vhd(self, vm_name, resource_group=None):
         return vhd_endpoint
 
     def current_ip_address(self, vm_name, resource_group=None):
-        # Returns first active IPv4 IpAddress only
+        """
+        Returns first active IPv4 IpAddress only.
+
+        To get IP we have to fetch:
+                              - nic object from VM
+                              - ip_config object from nic
+                              - public_ip object from ip_config
+        *_id - is a valid Azure resource id
+        e.g. - /subscriptions/<subscription>/resourceGroups/<resource_group>/providers/
+        Microsoft.Network/publicIPAddresses/<object_name>
+        """
+
         resource_group = resource_group or self.resource_group
         vm = self.vms_collection.get(resource_group_name=resource_group,
                                      vm_name=vm_name)
-        first_vm_if = vm.network_profile.network_interfaces[0]
-        if_name = os.path.split(first_vm_if.id)[1]
-        public_ip = self.network_client.public_ip_addresses.get(resource_group, if_name)
+        # Getting id of the first network interface of the vm
+        first_vm_if_id = vm.network_profile.network_interfaces[0].id
```

Comment:
I think this potential `AttrbuteError` should be handled better.

This is an example with a `vm_name` returned by `AzureSystem.list_vm()`, an expected use case for this method.
```
In [6]: az.mgmt.current_ip_address(vm_name='sbulage-16refs')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-a95be775f729> in <module>()
----> 1 az.mgmt.current_ip_address(vm_name='sbulage-16refs')

/home/mshriver/repos/wrapanapi/wrapanapi/msazure.py in current_ip_address(self, vm_name, resource_group)
    325                                                                                     ip_config_name)
    326         # Getting public IP id from the IP configuration object
--> 327         pub_ip_id = ip_config_obj.public_ip_address.id
```